atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+                property_dict[f'{val_name}_magnitude'] = value_magnitude
+
+        workflow_properties_dict = {}
+        for observable_type, observable_dict in observables.items():
+            flag_known_property = False
+            if observable_type in properties_known:
+                property_type_key = observable_type
+                property_type_value_key = properties_known[observable_type]
+                flag_known_property = True
+            property_dict = {property_type_value_key: []}
+            property_dict['label'] = observable_type
+            for key, observable in observable_dict.items():
+                property_values_dict = {'label': key}
+                for quant_name, val in observable.items():
+                    if quant_name == 'val':
+                        continue
+                    if quant_name == 'bins':
+                        continue
+                    if quant_name in property_keys_list:
+                        property_dict[quant_name] = val
+                    if quant_name in property_value_keys_list:
+                        property_values_dict[quant_name] = val
+                    # TODO Still need to add custom values here.
+
+                val = observable.get('value')
+                populate_property_dict(property_values_dict, 'value', val, flag_known_property=flag_known_property)
+                bins = observable.get('bins')
+                populate_property_dict(property_values_dict, 'bins', bins, flag_known_property=flag_known_property)
+                property_dict[property_type_value_key].append(property_values_dict)
+
+            if workflow_properties_dict.get(property_type_key):
+                workflow_properties_dict[property_type_key].append(property_dict)
+            else:
+                workflow_properties_dict[property_type_key] = [property_dict]
+
+        return workflow_properties_dict
+
+    def parse_workflow(self):
+
+        workflow_parameters = self._parameter_info.get('workflow').get('molecular_dynamics')
+        # TODO should store parameters that do not match the enum vals as x_h5MD params,
+        # not sure how with MDParser??
+        if workflow_parameters is None:
+            return
+
+        workflow_results = {}
+        ensemble_average_observables = self._observable_info.get('ensemble_average')
+        ensemble_property_dict = {
+            'property_type_key': 'ensemble_properties',
+            'property_type_value_key': 'ensemble_property_values',
+            'properties_known': {'radial_distribution_functions': 'radial_distribution_function_values'},
+            'property_keys_list': EnsembleProperty.m_def.all_quantities.keys(),
+            'property_value_keys_list': EnsemblePropertyValues.m_def.all_quantities.keys()
+        }
+        workflow_results.update(
+            self.get_workflow_properties_dict(ensemble_average_observables, **ensemble_property_dict)
+            )
```

Comment:
overindented )

Reply:
Like this:
```
        workflow_results.update(
            self.get_workflow_properties_dict(ensemble_average_observables, **ensemble_property_dict)
        )
```
??

Reply:
yes, not sure if the linter will actually pick it up but just to be safe hehe.

Reply:
yeah, the ruff extension for vscode auto-fixes a lot of spacing and such, but doesn't catch this for some reason.